scripts/ida_toggles.py

```diff
@@ -42,22 +43,98 @@ def add_toggle_data(self, dump_file_path):
                 raise
         self._add_toggle_data(dump_contents)
 
+    def _get_toggle_type(self, row_data):
+        """
+        Assign toggle type to model types
+        """
+
+        # convert sql dump model name to annotation report toggle type name
+        # this is usually the case with just devstack data dump, particularly from ecommerce
+        if row_data['model'] == "waffle.flag":
+            toggle_type = ToggleTypes.WAFFLE_FLAG
+        elif row_data['model'] == "waffle.switch":
+            toggle_type = ToggleTypes.WAFFLE_SWITCH
+        elif row_data['model'] == "waffle.sample":
+            toggle_type = ToggleTypes.WAFFLE_SAMPLE
+        else:
+            try:
+                toggle_type = ToggleTypes(row_data['model'])
+            except:
+                LOGGER.warning(
+                'Name of model not recognized: {}'.format(row_data['model'])
+                )
+                toggle_type = ToggleTypes.UNKNOWN
+        return toggle_type
+
+    def _handle_course_override_data(self, row_data):
+        """
+        Function to handle special case of Course overrides.
+        """
+        toggle_name = row_data['fields']['waffle_flag']
+        toggle_type = ToggleTypes.COURSE_WAFFLE_FLAG
+        toggle_data = row_data['fields']
+        if toggle_type not in self.toggles.keys():
+            self.toggles[toggle_type] = {}
+        toggle = self.toggles[toggle_type].get(toggle_name, None)
+        
+        if toggle is None:
+            # if a "CourseWaffleFlag/CourseOverride" was found in data before its corresponding
+            # CourseWaffleFlag, create an empty toggle to add course override data
```

Comment:
This is confusing.  A CourseWaffleFlag encapsulates a "waffle flag" and "course override".  I _think_ you mean:
```
            # if a CourseWaffleFlag's course override data was found before its corresponding
            # waffle flag, create an empty toggle to add course override data
```